Userland/Utilities/ldd.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/DeprecatedString.h>
+#include <AK/LexicalPath.h>
+#include <AK/StringBuilder.h>
+#include <AK/StringView.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <LibCore/MappedFile.h>
+#include <LibCore/System.h>
+#include <LibELF/DynamicLinker.h>
+#include <LibELF/DynamicLoader.h>
+#include <LibELF/DynamicObject.h>
+#include <LibELF/Image.h>
+#include <LibELF/Validation.h>
+#include <LibMain/Main.h>
+#include <ctype.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+static Vector<DeprecatedString> found_libraries;
+
+static ErrorOr<void> recusively_resolve_all_necessary_libraries(StringView interpreter_path, size_t recursive_iteration_max, size_t recusrive_iteration, ELF::DynamicObject& object)
+{
+    if (recusrive_iteration > recursive_iteration_max)
+        return ELOOP;
+
+    Vector<DeprecatedString> libraries;
+    object.for_each_needed_library([&libraries](StringView entry) {
+        libraries.append(DeprecatedString::formatted("{}", entry));
+    });
+    for (auto& library_name : libraries) {
+        auto possible_library_path = ELF::DynamicLinker::resolve_library(library_name, object);
```

Comment:
@ADKaster I don't see how we could drift away from LibELF resolution mechanism given that we use this function here and could also use it in a non-loading resolver?